### PR TITLE
Documentation improvements

### LINKED ISSUE #133: Improve documentation with missing info and better readability
```[tasklist]
### Tasks
- [x] overhaul documentation for better readability
- [x] do we want to use pyexasol for the connection in the user guide? looks cleaner imo
- [x] also info about adding SSL certificate?
- [x] mention each model is 2 part model and transformer (in dev guide at least)
- [ ] in dev docu: add info about change of model if parameters changed(happens in base_model_udf), and how
- [ ] in user docu: mention udf can change udf/model if the input set changes the model parameters
- [ ] add example usages in user docu (will be done in #164)
- [x] mention trigger slc dowload test after release to make sure it works with new version. test_language_container_deployer_cli_by_downloading_container (maybe trigger automatically in nox session which does a release)
- [x] add information about the certificates as explained [here](https://github.com/exasol/transformers-extension/pull/150/files/99549de939e855f1698a05a4145d3373f51ffab9#r1392322048 )
- [x] add explanation about model download and uplaod, details [here](https://github.com/exasol/transformers-extension/pull/150#discussion_r1392443311)
- [x] mention after release check .dist and .dist in language container for multiple wheels
- [x] mention windows not supported
- [x] change description of project to mention Huggingface
- [x] see comments in https://github.com/exasol/transformers-extension/pull/150
- [x] link to udf docu
- [x] link to bucketfs docu
- [ ] add screenshot from exa operations where to find parameters like bucketfs name (see screenshot) #164
- [x] mention pypi release and pip install
- [x] get correct link  to pypi and install comand
```

### LINKED ISSUE #188: Improve description of BucketFS Con
Add a note that, the path in bucket can't be empty.

https://github.com/exasol/transformers-extension/blob/5f6926305b48eecae0b6dbf0b4927fd0c0b3e127/doc/user_guide/user_guide.md?plain=1#L54

### PR DESCRIPTION
**Pull request by MarleneKress79789**
### All Submissions:

* [x] Is the title of the Pull Request correct?
* [x] Is the title of the corresponding issue correct?
* [x] Have you updated the changelog?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../../pulls) for the same update/change?  
* [x] Are you mentioning the issue which this PullRequest fixes ("Fixes...")
* [ ] Before you merge don't forget to run tests in AWS CodeBuild, by adding `[CodeBuild]` to the commit message
* Fixes #133 , 
* Fixes #188

### CODE REVIEW

**Review by tkilias — CHANGES_REQUESTED** [review]
See my comments and please change the target branch to the dev branch

File: pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [tool.poetry]
 name = "exasol-transformers-extension"
 version = "1.0.0"
-description = "An Exasol extension to use state-of-the-art pretrained machine learning models via the transformers api."
+description = "An Exasol extension for using state-of-the-art pretrained machine learning models via the Huggingface transformers api."
```
Comment (tkilias) [inline pyproject.toml:4]:
```suggestion
description = "An Exasol extension for using state-of-the-art pretrained machine learning models via the Hugging Face Transformers api."
```

File: buildspec.yml [hunk 2]
```diff
@@ -22,6 +22,5 @@ phases:
       - echo "$DOCKER_PASSWORD" | docker login --username "$DOCKER_USER" --password-stdin
   build:
     commands:
-      - poetry run nox -s unit_tests
```
[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply buildspec.yml:25]: we already run them in the github ci and running them twice for every commit seems wasteful.

File: doc/developer_guide/developer_guide.md [hunk 3]
```diff
@@ -126,3 +134,21 @@ Currently, the CodeBuild project is managed manually and is triggered with a web
 For this our aws-ci user is added to this Repository. The webhook can be configured in the AWS CodeBuild 
 project directly.
 The CodeBuild project also uses our DockerHub user for the build. For this it has access to the AWS SecretsManager.
+
+
+#### 3. Release download test
+
+After you do a Release on the project, you may want to trigger the slc dowload test ***link***
```
[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply doc/developer_guide/developer_guide.md]: fixed

File: doc/user_guide/user_guide.md [hunk 4]
```diff
@@ -111,10 +156,17 @@ Transformers Extension Package. See [the latest release](https://github.com/exas
...
+
+**Note:** The `PATH_IN_BUCKET` can not be empty.
+
 The `--ssl-cert-path` is optional if your certificate is not in the OS truststore. 
+This certificate is basically a list of trusted CA. It is needed for the server's certificate 
+validation by the client.
 The option `--use-ssl-cert-validation`is the default, you can disable it with `--no-use-ssl-cert-validation`.
 Use caution if you want to turn certificate validation off as it potentially lowers the security of your 
 Database connection.
+This is not to be confused with the client's own certificate. It may or may not include the private key.
```
[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply doc/user_guide/user_guide.md:168]: i trusted what mikhail wrote after he experimented with the TE in ai-lab https://github.com/exasol/transformers-extension/pull/150/files/99549de939e855f1698a05a4145d3373f51ffab9#r1392322048

File: doc/user_guide/user_guide.md [hunk 5]
```diff
@@ -273,11 +340,29 @@ models from the local filesystem into BucketFS:
...
+These should have been saved using transformers [save_pretrained](https://huggingface.co/docs/transformers/v4.32.1/en/installation#fetch-models-and-tokenizers-to-use-offline) 
+function to ensure proper loading by the Transformers Extension UDFs.
+You can download the model using python lke this:
+
+```python
+    for model_factory in [transformers.AutoModel, transformers.AutoTokenizer]:
+        # download the model an tokenizer from huggingface
+        model = model_factory.from_pretrained(model_name, cache_dir=<your cache path> / <huggingface model name>)
+        # save the downloaded model using the save_pretrained fuction
+        model.save_pretrained(<save_path> / "pretrained" / <model_name>)
```
[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply doc/user_guide/user_guide.md:354]: changed

File: doc/user_guide/user_guide.md [hunk 6]
```diff
@@ -273,11 +340,29 @@ models from the local filesystem into BucketFS:
...
+**Note:**  The `--path-in-bucket` can not be empty.
+**Note**: The options --local-model-path needs to point to a path which contains the model and its tokenizer. 
+These should have been saved using transformers [save_pretrained](https://huggingface.co/docs/transformers/v4.32.1/en/installation#fetch-models-and-tokenizers-to-use-offline) 
+function to ensure proper loading by the Transformers Extension UDFs.
+You can download the model using python lke this:
+
+```python
+    for model_factory in [transformers.AutoModel, transformers.AutoTokenizer]:
+        # download the model an tokenizer from huggingface
+        model = model_factory.from_pretrained(model_name, cache_dir=<your cache path> / <huggingface model name>)
```
[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply doc/user_guide/user_guide.md:352]: changed

File: doc/user_guide/user_guide.md [hunk 7]
```diff
@@ -203,35 +262,46 @@ There are two ways to install the language container: (1) using a python script
 
       ```sql
       ALTER SESSION SET SCRIPT_LANGUAGES=\
-      <ALIAS>=localzmq+protobuf:///<BUCKETFS_NAME>/<BUCKET_NAME>/<PATH_IN_BUCKET><CONTAINER_NAME>/?\
-              lang=<LANGUAGE>#buckets/<BUCKETFS_NAME>/<BUCKET_NAME>/<PATH_IN_BUCKET><CONTAINER_NAME>/\
+      _PYTHON3_TE_=localzmq+protobuf:///<BUCKETFS_NAME>/<BUCKET_NAME>/<PATH_IN_BUCKET><CONTAINER_NAME>/?\
```
[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply doc/user_guide/user_guide.md:265]: yes. but if we want to mention that we also need to mention to change it [here](https://github.com/exasol/transformers-extension/blob/main/language_container/exasol_transformers_extension_container/flavor_base/language_definition), right? or is this auto generated and i am not seeing where?

File: doc/developer_guide/developer_guide.md [hunk 8]
```diff
@@ -126,3 +134,21 @@ Currently, the CodeBuild project is managed manually and is triggered with a web
...
 project directly.
 The CodeBuild project also uses our DockerHub user for the build. For this it has access to the AWS SecretsManager.
+
+
+#### 3. Release download test
+
+After you do a Release on the project, you may want to trigger the slc dowload test ***link*** 
+to make sure the new SLC is uploaded and correctly named. 
+**This is especially important if the naming convention of the SLC was changed!**
+* filepath: tests/integration_tests/with_db/deployment/test_language_container_deployer_cli.py
```
[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply doc/developer_guide/developer_guide.md:144]: chnaged

File: doc/user_guide/user_guide.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (tkilias) [reply doc/user_guide/user_guide.md]: good question, lets keep it as for the moment, maybe revisit this later

File: doc/user_guide/user_guide.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (tkilias) [reply doc/user_guide/user_guide.md:168]: ok, then keep it, and I need to check later if we support it

File: buildspec.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tkilias) [reply buildspec.yml:25]: ok

File: doc/user_guide/user_guide.md [hunk 9]
```diff
@@ -203,35 +262,46 @@ There are two ways to install the language container: (1) using a python script
...
+Please make sure that the selected models are in the `Pytorch` model library section.
 
 ### 1. Model Downloader UDF
 Using the `TE_MODEL_DOWNLOADER_UDF` below, you can download the desired model 
-from the huggingface hub and upload it to bucketfs.
+from the huggingface hub and upload it to BucketFS.
+This requires the Exasol Database to have internet access, since the UDF will 
+download the model from Huggingface to the Database without saving it somewhere else intermittently.
+If you are using the Exasol DockerDB, this is not the case by default, and you need to specify a name server. 
+For example `--nameserver 8.8.8.8` will set it to use Google DNS.
```
[the remark this answers is not in the document]
Comment (MarleneKress79789) [reply doc/user_guide/user_guide.md]: changed

**Review by tkilias — APPROVED** [review]

**Pull request merged**